### PR TITLE
improves form validation UX for create service form

### PR DESCRIPTION
**Pull request by allenjprice**
still only verifies that the `name` and `description` are not empty, and that the `code` is neither empty nor used by any other service

### CODE REVIEW

**Comment by allenjprice** [comment]
this is WIP - it's only evaluating service _creation_ and not service _edits_. That should be similar, but involves removing the check for codes (since you can't edit a service's code!)

**Comment by allenjprice** [comment]
@jalehman @queenieho this is ready for review now, promise!

File: src/cljs/admin/services/events.cljs [hunk 1]
```diff
@@ -384,7 +385,50 @@
  [(path db/path)]
  (fn [db _]
    (-> (update db :form dissoc :name :description :code :properties :catalogs :price :cost :rental :fields)
-       (assoc :form db/form-defaults))))
+       (assoc :form db/form-defaults)
+       (dissoc :form-validation)
```
Comment (jalehman) [inline src/cljs/admin/services/events.cljs]: Why `dissoc` and then `assoc`? The `assoc` would override whatever was there anyway.

File: src/cljs/admin/services/events.cljs [hunk 2]
```diff
@@ -384,7 +385,50 @@
...
+ (fn [{db :db} _]
+   (let [name        (get-in db [:form :name])
+         description (get-in db [:form :description])
+         code        (get-in db [:form :code])]
+     {:db       (-> (assoc-in db [:form-validation :name] (not (empty? name)))
+                    (assoc-in [:form-validation :description] (not (empty? description)))
+                    (assoc-in [:form-validation :code] (not (or (empty? code)
+                                                                 (contains? (set (map :code (norms/denormalize db :services/norms))) code)))))
+      :dispatch [::validate-create]})))
+
```
Comment (jalehman) [inline src/cljs/admin/services/events.cljs]: newline

File: src/cljs/admin/services/subs.cljs [hunk 3]
```diff
@@ -83,3 +83,11 @@
  :<- [:services/list]
  (fn [services _]
    (vec (distinct (mapcat :catalogs services)))))
+
+
+
+(reg-sub
+ :service.form/is-valid?
+ :<-[db/path]
```
Comment (jalehman) [inline src/cljs/admin/services/subs.cljs]: `:<- [db/path]` (space between '-' and '[')

**Comment by allenjprice** [comment]
@jalehman changes submitted 💃

**Pull request merged**